### PR TITLE
Resolve remaining AWS TODOs

### PR DESCRIPTION
**Pull request by mbarkley**

### CODE REVIEW

**Comment by chaopeng** [comment]
lgtm on me

File: lib/aws/warehouse.go [hunk 1]
```diff
@@ -775,8 +775,13 @@ func (wh *AccountWarehouse) ensureLoginProfile(userName string) (string, error)
...
 	// garbage collection call
-	makeRoom := princSpec.params.ManagedKeysPerAccount - 1
-	keyTTL := timeutil.KeyTTL(princSpec.params.MaxKeyTTL, princSpec.params.ManagedKeysPerAccount)
+	keysPerAccount := princSpec.params.ManagedKeysPerAccount
+	if keysPerAccount < 1 {
+		return nil, fmt.Errorf("cannot create access key: maximum number keys per account is %d", keysPerAccount)
+	}
+
+	makeRoom := keysPerAccount - 1
+	keyTTL := timeutil.KeyTTL(princSpec.params.MaxKeyTTL, keysPerAccount)
```
Comment (chaopeng) [inline lib/aws/warehouse.go:784]: Please add a test ensure ManageAccountKeys respect keysPerAccount and TTL.

Comment (mbarkley) [reply lib/aws/warehouse.go:784]: We have [some tests](https://github.com/GoogleCloudPlatform/healthcare-federated-access-services/blob/master/lib/aws/warehouse_test.go#L260-L386) for this from a previous PR. Do they cover what you were looking for?

Comment (chaopeng) [reply lib/aws/warehouse.go:784]: It is ok. Thank you for pointing.

**Pull request closed without merging**